src/util/misc/verify.ts

```diff
@@ -0,0 +1,31 @@
+import { ComponentEmojiResolvable, Guild } from "discord.js";
+import fetch from "node-fetch";
+
+export function isNotEmpty(value: string): boolean {
+	return value !== "";
+}
+
+export async function emojiExists(guild: Guild, emoji: ComponentEmojiResolvable): Promise<boolean> {
+	return (await fetch(`https://cdn.discordapp.com/emojis/${emoji}`)).ok;
+}
+
+export async function roleExists(guild: Guild, role: string): Promise<boolean> {
+	return await guild.roles.fetch(role).then((role) => role !== null);
+}
+
+export async function channelExists(guild: Guild, channel: string): Promise<boolean> {
+	return await guild.channels.fetch(channel).then((channel) => channel !== null);
+}
```

Comment:
Bot guilds and channels should have a catch for errors

Reply:
on it

Reply:
ds.js sucks when it comes to fetching...
some throw an error, some just return null its so confusing

Reply:
As far as I'm aware, fetch always errors.

Reply:
sadly not... guild.roles.fetch() just returns null with no error. guild.channels.fetch() is supposed to return the channel or null but just throws an error 
check the docs